server/tests/integration/test_api.py

```diff
@@ -517,6 +517,152 @@ def test_create_two_entities(self):
+        self.assertTrue('assignment' in tokens[0])
+        self.assertTrue('Hog Project' in tokens[0])
+
+    def test_tokenize_single_quoted_string(self):
+        """ Tests that args with spaces can be grouped by single quotations """
+        query = "-assignment 'Hog Project'"
+        tokens = self.API.tokenize(query)
+        self.assertTrue('assignment' in tokens[0])
+        self.assertTrue('Hog Project' in tokens[0])
+
+    def test_multiple_flags(self):
+        """ Tests that the multiple groups can be parsed together. """
+        query = '-assignment Hog -date --before 2015-06-22'
+        tokens = self.API.tokenize(query)
+        self.assertTrue('assignment' in tokens[0])
+        self.assertTrue('Hog' in tokens[0])
+        self.assertTrue('date' in tokens[1])
+        self.assertTrue('before' in tokens[1])
+        self.assertTrue('2015-06-22' in tokens[1])
+        
+    def test_datatype_int(self):
+        """ Tests that integers are parsed normally. """
+        query = '-price 10'
+        tokens = self.API.tokenize(query)
+        self.assertTrue('price' in tokens[0])
+        self.assertTrue('10' in tokens[0])
+        
+    def test_datatype_boolean(self):
+        """ Tests that booleans are parsed normally. """
+        query = '-pricey true'
+        tokens = self.API.tokenize(query)
+        self.assertTrue('pricey' in tokens[0])
+        self.assertTrue('true' in tokens[0])
+
+    def test_single_character_arg(self):
+        """ Tests that an arg with only length one can be parsed normally. """
+        query = '-price 1'
+        tokens = self.API.tokenize(query)
+        self.assertTrue('price' in tokens[0])
+        self.assertTrue('1' in tokens[0])
+        
+    def test_laziness_with_quotations(self):
+        """ Tests that quotations are lazily matched """
+        query = '-date --before "2015-06-22" -assignment "Homework 1"'
+        tokens = self.API.tokenize(query)
+        self.assertTrue('date' in tokens[0])
+        self.assertTrue('before' in tokens[0])
+        self.assertTrue('2015-06-22' in tokens[0])
+        self.assertTrue('assignment' in tokens[1])
+        self.assertTrue('Homework 1' in tokens[1])
+        
+        
+    #######################
+    # TRANSLATE OPERATORS #
+    #######################
+
+    def test_translate_basic(self):
+        """ Basic extraction of arg and operator """
+        query = '-date --before 2015-06-22'
+        scope = self.API.translate(query)
+        operator, arg = scope['date']
+        self.assertEqual(operator, op.__lt__)
+        self.assertEqual(arg, '2015-06-22')
+        
+    def test_translate_operators(self):
+        """ Translate to eq if no operator specified """
+        query = '-assignment Scheme'
+        scope = self.API.translate(query)
+        operator, arg = scope['assignment']
+        self.assertEqual(operator, op.__eq__)
+        self.assertEqual(arg, 'Scheme')
+
+    ###########
+    # QUERIFY #
+    ###########
```

Comment:
This is a good start on testing the querying process. If we want to take it further, we would want to verify the results.